citesphere/src/main/java/edu/asu/diging/citesphere/web/user/GilesDocumentController.java

```diff
@@ -60,4 +73,72 @@ public void get(HttpServletResponse response, @PathVariable String itemId, @Path
         }
 
     }
+    
+    public void setFileNameAndContentType(List<IGilesUpload> uploadOptionalList, String fileId) {
```

Comment:
This is all very convoluted. I think it would be a lot more elegant if you use the `generate` function of the stream api and generate a list of `GilesFile`s that you then filter for the given file id. E.g.: https://www.javacodegeeks.com/2014/08/java-yield-like-using-stream-api.html
the `generateRandomIntStream` method would return a stream that is created by iterating over all gilesUploads and then over all files and pages in there. The resulting stream can then be filtered in for the desired fileId. Does that make sense?